### PR TITLE
Update CI scripts and apps

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**
Changes proposed in this pull request:

- change Electron to Chrome on both Travis and local environment
- launch Chrome in headless mode on Travis. Sadly, it's impossible to be done for local env
- `npm run e2e:run` and `npm run e2e:open` behave differently although they both run Chrome. All errors shown on Travis will also be visible locally using the first command while they might be invisible using the second one.
- _Cypress_ and _node_modules_ are now cached which gives a significant speedup 🚀 💪
- a loooong research has been done about "stabilizing" (fixing) tests that are randomly failing on Travis. Without success :(
- disable video recording (another small speedup 🚀 💪) 
- give those infamous "unstable" test cases a second chance using `cypress-plugin-retries`. **Please use this tool wisely in the future**
- other tiny updates on CI scripts to make them faster

**Related issue(s)**

### CODE REVIEW

File: core/examples/luigi-sample-angular/cypress.json [hunk 1]
```diff
@@ -8,5 +8,6 @@
   "chromeWebSecurity": false,
   "viewportWidth": 1250,
   "viewportHeight": 790,
-  "baseUrl": "http://localhost:4200/"
+  "baseUrl": "http://localhost:4200/",
+  "numTestsKeptInMemory": 0
```
Comment (marynaKhromova) [inline core/examples/luigi-sample-angular/cypress.json]: Are you sure we don't need to keep any? By default it's 50, maybe it's better just to reduce the number.

File: core/examples/luigi-sample-angular/e2e/tests/navigation.spec.js [hunk 2]
```diff
@@ -69,36 +69,42 @@ describe('Navigation', () => {
...
         // wrap this body with cy so as to do cy actions inside iframe elements
         cyIframe = cy.wrap(iframeBody);
         //now you can forget about that you are in iframe. you can do necessary actions finding the elements inside the iframe
-        // {cyElement is the cypress object here}
         cyIframe
-          .find('.fd-list-group__item strong')
+          .find('.fd-list-group__item')
           .contains('keepSelectedForChildren')
           .click();
+        cy.wait(500);
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/e2e/tests/navigation.spec.js]: I tried to remove it completely, it was failing about 90% times.

File: core/examples/luigi-sample-angular/e2e/tests/luigi-client-ux-manager-features.spec.js [hunk 3]
```diff
@@ -70,15 +70,17 @@ describe('Luigi client ux manger features', () => {
     });
 
     it('loading indicator', () => {
+      Cypress.currentTest.retries(3);
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/e2e/tests/luigi-client-ux-manager-features.spec.js]: I think we should keep it individual for each test, based on its _probability to fail_. As I mentioned in the description, it's important to be really really careful with these retries.

File: core/examples/luigi-sample-angular/cypress.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/cypress.json]: I'll try to remove it completely, let's see what happens ;)

File: core/examples/luigi-sample-angular/e2e/support/commands.js [hunk 4]
```diff
@@ -51,8 +51,8 @@ const isHashRoutingOn = () => {
   return useHashRouting;
 };
 
-Cypress.Commands.add('expectPathToBe', pathWithoutHash =>
-  cy.location().should(location => {
+Cypress.Commands.add('expectPathToBe', (pathWithoutHash, timeout = undefined) =>
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/e2e/support/commands.js]: I like to keep things clear for a developer. It's implicitly said that this param is optional.

File: .travis.yml [hunk 5]
```diff
@@ -1,19 +1,30 @@
 language: node_js
 node_js:
-  - 'node'
+  - '11.14' #TODO: change it to 'node' once Travis supports the newest version
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply .travis.yml]: ... in the future :) 
btw. I'll check, maybe they fixed it.

File: core/examples/luigi-sample-angular/e2e/tests/luigi-client-link-manager-features.spec.js [hunk 6]
```diff
@@ -8,6 +8,7 @@ describe('Luigi client features', () => {
   });
 
   it('linkManager features', () => {
+    Cypress.currentTest.retries(2);
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/e2e/tests/luigi-client-link-manager-features.spec.js]: I based it on probability of failing for each test. This one was failing significantly less frequently than the others.

File: test/e2e.sh [hunk 7]
```diff
@@ -2,11 +2,19 @@
 
 pwd
 
-echo Installing lerna
-npm install -g lerna
+cd core/examples/luigi-sample-angular
+pwd
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply test/e2e.sh]: Not really, I was experimenting with installing more dependencies locally but I succeeded only with Cypress.

File: core/examples/luigi-sample-angular/e2e/tests/navigation.spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/e2e/tests/navigation.spec.js]: I also tried to set it to `2000` but it was still failing. I think `500` is a good compromise.

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (parostatkiem) [reply .travis.yml]: Sadly, 
image

File: core/examples/luigi-sample-angular/cypress.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/cypress.json]: Nothing changed, still works 👍

**Pull request merged**